game-core/src/test/java/games/strategy/triplea/delegate/PacificTest.java

```diff
@@ -108,20 +108,44 @@ void testNonJapanAttack() {
     // Defending US infantry
     DiceRoll roll =
         DiceRoll.rollDice(
-            infantryUs, true, americans, bridge, mock(IBattle.class), "", territoryEffects, null);
+            infantryUs,
+            true,
+            americans,
+            bridge,
+            mock(IBattle.class),
+            "",
+            territoryEffects,
+            null,
+            infantryUs);
```

Comment:
It's striking we have the same parameter passed in twice to a method. IMHO it is a pretty significant API code smell. Perhaps an extract parameter object refactor would fix that?

IMO the existing API already has significant technical debt, I don't think that is fair reason to add more potential debt for someone else to attempt to fix. It's also very notable that a number of other parameters are already useless, seems like this API is already stretched.

Reply:
I think you are looking at outdated code here as well.